### PR TITLE
Fix ErrorMessages serialization for AOT

### PR DESCRIPTION
**Pull request by jevansaks**
When testing out real AOT, I uncovered one suppression that was covering a real issue. This call to Serialize was passing a custom JsonSerializerOptions that didn't include the TypeInfoResolver. Adding that revealed that keywords weren't serializing without a couple of extra primitive types that they depended on (`decimal`, `uint`, `JsonNode[]`, etc).

This revealed that we weren't including the custom TypeInfos for keywords during that serialization either, so to fix that we need to construct a JsonSerializerOptions that has the chain from our own as well as any callers.

I've changed the pattern so that we call serialize/deserialize with a custom Options now that's based on our source generated context but can dynamically get the context from any dynamically registered keywords too.

### CODE REVIEW

File: JsonSchema/SchemaKeywordRegistry.cs [hunk 1]
```diff
@@ -21,78 +22,82 @@ namespace Json.Schema;
 public static class SchemaKeywordRegistry
 {
 	private static readonly ConcurrentDictionary<string, Type> _keywords;
-	private static readonly ConcurrentDictionary<Type, JsonTypeInfo> _keywordTypeInfos;
+	private static readonly ConcurrentDictionary<Type, JsonSerializerContext> _keywordTypeInfoResolvers = new();
```
Comment (gregsdennis) [inline JsonSchema/SchemaKeywordRegistry.cs:25]: It doesn't look like this is ever initialized with the default keyword set.

File: JsonSchema/JsonSchema.cs [hunk 2]
```diff
@@ -91,12 +92,96 @@ public class JsonSchema : IBaseDocument
...
+					TypeInfoResolver = _typeInfoResolver
+#endif
+				};
+
+				return _serializerOptions!;
+			}
+		}
+	}
+
+	internal static JsonSerializerOptions SerializerOptionsUnsafeRelaxedJsonEscaping
```
Comment (gregsdennis) [inline JsonSchema/JsonSchema.cs:114]: It doesn't look like these things actually have to do with the `JsonSchema` class specifically.  I'd like them removed into a separate (static) class, please.

File: JsonSchema/SchemaKeywordRegistry.cs (hunk 1 above)
Comment (jevansaks) [reply JsonSchema/SchemaKeywordRegistry.cs:25]: It isn't, those are from the default JsonSchema serializer context. I can add a comment.

File: JsonSchema/JsonSchema.cs (hunk 2 above)
Comment (jevansaks) [reply JsonSchema/JsonSchema.cs:114]: Sure.

File: JsonSchema/SchemaKeywordRegistry.cs (hunk 1 above)
Comment (gregsdennis) [reply JsonSchema/SchemaKeywordRegistry.cs:25]: Okay... this is importing new contexts.  Got it.

File: JsonSchema/JsonSchema.cs (hunk 2 above)
Comment (jevansaks) [reply JsonSchema/JsonSchema.cs:114]: Would you be ok if they're on the JsonSchemaSerializerContext?

Comment (gregsdennis) [reply JsonSchema/JsonSchema.cs:114]: Yeah, if that won't interfere with anything else.

**Comment by gregsdennis** [comment]
Looks great!

**Pull request merged**